packages/core/tests/controllers/SIWEController.test.ts

```diff
@@ -0,0 +1,63 @@
+import { describe, expect, it } from 'vitest'
+import { SIWEController, SIWEStatus, type SIWEControllerClient } from '../../index.js'
+
+// -- Mocks -------------------------------------------------------------
+const session = { address: '0x', chainId: 1 }
+const client: SIWEControllerClient = {
```

Comment:
You don't need explicit `: SIWEControllerClient` here. If you were to pass invalid type to `SIWEController.setSIWEClient` that's where type error should appear.